priv/cabbage/apps/itest/lib/reorg.ex

```diff
@@ -0,0 +1,121 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule Itest.Reorg do
+  @moduledoc """
+    Chain reorg triggering logic.
+  """
+
+  alias Itest.Poller
+
+  require Logger
+
+  @node1 "geth-1"
+  @node2 "geth-2"
+
+  @rpc_nodes ["http://localhost:9000", "http://localhost:9001"]
+
+  def execute_in_reorg(func) do
+    if Application.get_env(:cabbage, :reorg) do
+      pause_container!(@node1)
+      unpause_container!(@node2)
+
+      Process.sleep(10_000)
+
+      func.()
+
+      Process.sleep(10_000)
+
+      pause_container!(@node2)
+      unpause_container!(@node1)
+
+      Process.sleep(30_000)
+
+      response = func.()
+
+      Process.sleep(30_000)
+
+      unpause_container!(@node2)
+      unpause_container!(@node1)
+
+      :ok = Poller.wait_until_peer_count(1)
+
+      Process.sleep(20_000)
+
+      response
```

Comment:
I don't like these `sleeps`, because they're undeterministic. Is there a way for us to make this less prone to error?

Reply:
I can request the block number from nodes and wait for 5 generated blocks. I'll do this.